### PR TITLE
POC: Open GitHub PR from dashboard drawer

### PR DESCRIPTION
**Pull request by Clarity-89**
**What is this feature?**

Add a button to open a GH PR when saving dashboard. 

Screenshot 2024-11-25 at 18 04 14

### CODE REVIEW

File: public/app/features/dashboard-scene/saving/SaveProvisionedDashboard.tsx [hunk 1]
```diff
@@ -89,9 +103,11 @@ export function SaveProvisionedDashboard({ meta, drawer, changeInfo, dashboard }
           <Input {...register('path')} />
         </Field>
 
-        <Field label="Branch" description="Only supported by GitHub right now">
-          <Input {...register('ref')} />
-        </Field>
+        {isGitHub && (
+          <Field label="Branch" description="Only supported by GitHub right now">
```
Comment (MissingRoberto) [inline public/app/features/dashboard-scene/saving/SaveProvisionedDashboard.tsx:107]: Do you think that this clarification necessary? I don't think we will support branches for anything else in a very long time.

File: public/app/features/dashboard-scene/saving/SaveProvisionedDashboard.tsx [hunk 2]
```diff
@@ -43,7 +51,13 @@ export interface Props {
...
 export function SaveProvisionedDashboard({ meta, drawer, changeInfo, dashboard }: Props) {
   const [saveDashboard, request] = useUpdateRepositoryFilesMutation();
-  const { register, handleSubmit } = useForm({ defaultValues: getDefaultValues(meta) });
+  const defaultValues = getDefaultValues(meta);
+  const { register, handleSubmit, watch } = useForm({ defaultValues });
+  const repositoryConfigQuery = useGetRepositoryQuery({ name: defaultValues.repo });
+  const repositoryConfig = repositoryConfigQuery?.data?.spec;
+  const isGitHub = repositoryConfig?.type === 'github';
+  const [repo, ref] = watch(['repo', 'ref']);
+  const href = createPRLink(repositoryConfig, repo, ref);
```
Comment (ryantxu) [inline public/app/features/dashboard-scene/saving/SaveProvisionedDashboard.tsx:60]: rather just showing a link; I think this needs to be what happens *after* successfully writing to the branch.

File: public/app/features/dashboard-scene/saving/SaveProvisionedDashboard.tsx (hunk 1 above)
Comment (Clarity-89) [reply public/app/features/dashboard-scene/saving/SaveProvisionedDashboard.tsx]: Updated the description.

File: public/app/features/dashboard-scene/saving/SaveProvisionedDashboard.tsx (hunk 2 above)
Comment (Clarity-89) [reply public/app/features/dashboard-scene/saving/SaveProvisionedDashboard.tsx:60]: You mean do not show the link until the branch has some commits?

**Review by ryantxu — APPROVED** [review]
🤦🏻

**Pull request merged**